lib/catalog/martyrology.ts

```diff
@@ -3322,6 +3322,7 @@ export const Martyrology: { catalog: MartyrologyCatalog } = {
     },
     passion_of_saint_john_the_baptist: {
       name: 'Passion of Saint John the Baptist',
+      titles: [Title.Martyr],
```

Comment:
```suggestion
     // § 308.b. of the [GIRM](https://www.catholicculture.org/culture/library/view.cfm?recnum=337)
      titles: [Title.Martyr],
```

Reply:
will definitely approve once this is in

Reply:
If we add this comment here, we should add it to all entities with the title of Martyr. In my view, this doesn't make much sense because we are simply following the missal's rule.

However, since this saint is celebrated twice a year for different reasons (which is much less common and limited to a few saints), you make a good point about clarifying this case. We celebrate him as a martyr here, while for the Nativity of Saint John the Baptist, this is not the case (hence the celebration in white for that feast). So I'm adding a comment here. This way, in the future, we can avoid mistakenly adding the "Martyr" title to `nativity_of_john_the_baptist`.

Thank you @TobiTenno  🙏 .

Reply:
my point is that we should add it as we're modifying things so that we can add it when we add something and notice an inconsistency

Reply:
> If we add this comment here, we should add it to all entities with the title of Martyr.

As I see it, we should add a source reference to any and all data we use in romcal, regardless of the source type (first-party/second-party/third-party source), although first-party sources should be always prefered (like the Roman Missal, incl GIRM).